clippy_lints/src/assign_ops.rs

```diff
@@ -51,8 +51,10 @@ declare_clippy_lint! {
 /// **Why is this bad?** Most likely these are bugs where one meant to write `a
 /// op= b`.
 ///
-/// **Known problems:** Someone might actually mean `a op= a op b`, but that
-/// should rather be written as `a = (2 * a) op b` where applicable.
+/// **Known problems:** Clippy cannot know for sure if `a op= a op b` should have
+/// been `a = a op a op b` or `a = a op b`/`a op= b`. Therefore it suggests both.
+/// If `a op= a op b` is really the correct behaviour then
+/// rewrite it as `a = (2 * a) op b` as it's less confusing.
```

Comment:
I think I came up with `a = (2 * a) op b` and don't like it anymore :smile: Would be better to write this as `a = a op a op b`, I think.

Maybe formulate the second part a little different:

> If `a op= a op b` is really the correct behaviour it should be written as `a = a op a op b` as it's less confusing.

That sounds less like a suggestion, that you would expect in an error message and more like the thought behind this lint.